src/trezor/wire/__init__.py

```diff
@@ -209,3 +213,14 @@ def import_workflow(ctx, req, pkgname, modname, *args):
     await ctx.write(
         Failure(code=FailureType.UnexpectedMessage, message="Unexpected message")
     )
+
+
+def tezos_remove_handelrs():
+    # remove all handlers, except the ones used in tezos baking
+    for handler in workflow_handlers.keys():
+        if handler not in tezos_baking_allowed_messages:
+            workflow_handlers.pop(handler, None)
+
+
+def is_baking():
+    return len(workflow_handlers) == len(tezos_baking_allowed_messages)
```

Comment:
let's not do the check this way.

instead, i dunno, have a top-level `TEZOS_BAKING` constant that you set to `True` when entering baking mode.
but `homescreen.py` is probably more appropriate for it, given that it would be the only place that uses it.

(Also consider `workflow.closedefault` and `startdefault`. Perhaps making a custom homescreen layout would be a better solution? But I'm still unsure as to how exactly it would work. Just an option to keep in mind.)

Reply:
I think making a custom homescreen layout would solve the issue with the `show_baking_operation` and `show_endorsement_operation` screens (need to sleep the loop to show for the defined time). We could use the homescreen to show the last signed baking operation. But I'm still not sure how to make this work.